client/my-sites/domains/domain-search/site-redirect-step.jsx

```diff
@@ -1,90 +1,99 @@
 
 		return (
-			<div className="site-redirect-step">
-				<form className="site-redirect-step__form card" onSubmit={ this.handleFormSubmit }>
-
-					<div className="site-redirect-step__domain-description">
-						<p>
-							{ this.translate( 'Redirect {{strong}}%(domain)s{{/strong}} to this domain', {
-								components: { strong: <strong /> },
-								args: { domain: this.props.selectedSite.slug }
-							} ) }
-						</p>
-					</div>
-
-					<DomainProductPrice
-						price={ price }
-						requiresPlan={ false } />
-
-					<fieldset>
-						<input
-							className="site-redirect-step__external-domain"
-							type="text"
-							value={ this.state.searchQuery }
-							placeholder={ this.translate( 'Enter a domain', { textOnly: true } ) }
-							onChange={ this.setSearchQuery }
-							onClick={ this.recordInputFocus } />
-						<button className="site-redirect-step__go button is-primary"
-								onClick={ this.recordGoButtonClick }>
-							{ this.translate( 'Go', {
-								context: 'Upgrades: Label for adding Site Redirect'
-							} ) }
-						</button>
-					</fieldset>
-				</form>
+			<div>
+				<div className="domain-search__site-redirect-step">
+					<Card>
+						<form onSubmit={ this.handleFormSubmit }>
+
+							<div className="domain-search__site-redirect-step-description">
+								<p>
+									{ translate( 'Redirect {{strong}}%(domain)s{{/strong}} to this domain', {
+										components: { strong: <strong /> },
+										args: { domain: this.props.selectedSite.slug }
+									} ) }
+								</p>
+							</div>
+
+							<DomainProductPrice
+								price={ price }
+								requiresPlan={ false } />
+
+							<fieldset>
+								<FormTextInputWithAffixes
+									type="text"
+									value={ this.state.searchQuery }
+									prefix="http://"
+									placeholder={ translate( 'Enter a domain', { textOnly: true } ) }
+									onChange={ this.setSearchQuery }
+									onClick={ this.recordInputFocus } />
+							</fieldset>
+							<FormFooter>
+								<FormButton
+									onClick={ this.recordGoButtonClick }>
+									{ translate( 'Add Site Redirect', { context: 'button' } ) }
+								</FormButton>
+							</FormFooter>
+						</form>
+					</Card>
+				</div>
 			</div>
 		);
-	},
-
-	recordInputFocus: function() {
-		this.recordEvent( 'inputFocus', this.state.searchQuery );
-	},
+	}
 
-	recordGoButtonClick: function() {
-		this.recordEvent( 'goButtonClick', this.state.searchQuery );
-	},
+	recordInputFocus = () => {
+		this.props.recordInputFocus( this.state.searchQuery );
+	}
 
-	setSearchQuery: function( event ) {
-		this.setState( { searchQuery: event.target.value } );
-	},
+	recordGoButtonClick = () => {
+		this.props.recordGoButtonClick( this.state.searchQuery );
+	}
 
-	handleFormSubmit: function( event ) {
-		var domain;
+	setSearchQuery = ( event ) => {
+		// Removes the protocol part
+		const location = event.target.value.replace( /.*:\/\//, '' );
```

Comment:
I'd put something like `^\w*` at the beginning of that regexp instead to ensure that we only strip the protocol at the beginning of the location, but not if it's passed as a query argument or something later on in the string.
So, for `https://example.com`, `https://` should be stripped, but for `example.com/stuff?redirect_for_some_reason=https://foobar.com`, we should keep it as-is. I'm not even sure we support that (I think we do, but not fully, as in, it will probably break for non-trivial cases), but anyway, it should not hurt.